### PR TITLE
Request

### PR DESCRIPTION
**Pull request by suretrust**
Kindly review. According to rails 5 [release notes](http://rspec.info/blog/2016/07/rspec-3-5-has-been-released/#rails-support-for-rails-5), we used request specs for our controller. 

Also, we haven't completed the controller, but we wanted to be sure we're heading in the right direction.

### CODE REVIEW

File: config/routes.rb [hunk 1]
```diff
@@ -1,3 +1,24 @@
 Rails.application.routes.draw do
-  devise_for :users
+  devise_for :users, path: 'accounts', controllers: {
+    registrations: 'users/registrations'
+  }
+
+  root 'newsfeed#show'
+  
+  concern :likable do
```
Comment (rvvergara) [inline config/routes.rb]: Please enlighten me on this block of code here :) I think we can simply do this in the `posts` resources below:

```ruby
resources :likes, module: :posts, only: %i[create destroy index]
```
To make it simpler. If you're wondering why I used `modules` here -> See this [tutorial](https://gorails.com/episodes/comments-with-polymorphic-associations) to make your polymorphic associations much easier to reason about

File: app/controllers/users/registrations_controller.rb [hunk 2]
```diff
@@ -0,0 +1,27 @@
+# frozen_string_literal: true
+
+class Users::RegistrationsController < Devise::RegistrationsController
+  
+  def new
```
Comment (rvvergara) [inline app/controllers/users/registrations_controller.rb]: We are overriding all of `devise` 's methods here. Please refer to [this](https://github.com/plataformatec/devise#configuring-controllers) part of the `devise` documentation for details on configuring its controllers. When you override `devise` 's default actions, you are foregoing the benefits it provides you.

If fully overriding the registrations_controller is what you really want then keep this. But just confirm with the documentation about it first.

File: config/routes.rb (hunk 1 above)
Comment (suretrust) [reply config/routes.rb]: @rvvergara, thanks for the suggestion. 

We are using concern to write a DRYer code. if we were to use `module:`, there would be a lot of duplicates and our code may even require us to nest the route more than one level deep.

File: app/controllers/comments_controller.rb [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+  before_action :authenticate_user!
+  before_action :set_comment, only: [:update, :destroy]
+
+  def edit
+    @comment = Comment.find(params[:id])
+  end
+
+  def create
+    @post = Post.find(params[:post_id])
+    @comment = @post.comments.new comment_params
```
Comment (rvvergara) [inline app/controllers/comments_controller.rb]: Let's use `build` here instead of `new`

Comment (suretrust) [reply app/controllers/comments_controller.rb]: Could you please always tell us why? 

According to [rails active record explanation here,](https://github.com/rails/rails/blob/959fb8ea651fa6638aaa7caced20d921ca2ea5c1/activerecord/lib/active_record/relation.rb#L83-L85) `build` is just an alias for `new`.

**Comment by suretrust** [comment]
## User Interface

@rvvergara, we're not sure what went wrong, our best guess is perhaps, your local code base differs from what we have in our repo or you're in another branch. Attached below is our home page.

devium

**Comment by mosaaleb** [comment]
Hi @rvvergara, thanks for your tips, although It would more useful if you elaborate more on the these tips:
- > It isn't necessary to write views and request specs for this. What we actually need are feature specs.

why, and how it will be useful to write TDD tests after we have wrote the code?

- We have decided to divide `requests_controller.rb` into two controllers `outgoing` and `incoming`, as [**Separation of concerns**](https://en.wikipedia.org/wiki/Separation_of_concerns) where each controller will handle only one concern: `incoming` will handle *sending request* and *canceling request*, while `outgoing` will handle *rejecting a request*, this way we achieved more modular design and encapsulation that neither of the two controller knows how the other works since they're both performing different actions. That way we can safely add more logic to our app without thinking too much of where is this logic should reside.
- We have deviated a little from the standard requirements, and decided to put our touch :). We are fully aware that any user can visit any page (customer gaining strategy), but the user can't interact without authentication. So any user can perform any show action on any controller.

Thanks

**Comment by mosaaleb** [comment]
Hi @rvvergara,
- > TDD doesn't mean you cannot write the tests after you've written your code.

Sure it does, but does it follow the convention? It's clear from the name that your development is driven by your tests.

- We have already replaced the controller specs with request specs. The request specs add value not only to `API Testing`, but also to `Security Testing` which does not belong to feature specs. It's important to make sure that only a specific user can `delete` a post, or can `comment` on a post for example. Feature specs also cannot cover all the test case scenarios, as it is considered as high level tests.

- I have just cloned the repo two times, and the homepage working fine.

**Comment by suretrust** [comment]
Amended as requested.

**Comment by rvvergara** [comment]
Ok I guess the project is good to go!

# :+1: 

You may now merge with `master` and submit a completion form! :)

**Pull request merged**